### PR TITLE
Add slayer superior foe notification

### PR DESCRIPTION
**Pull request by xaviergmail**
Superiors will despawn after two minutes: https://twitter.com/JagexAsh/status/777629171027734528

Hopefully this helps a few people not miss out on a chance at those juicy drops

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerConfig.java [hunk 1]
```diff
@@ -55,6 +55,16 @@ default boolean showItemOverlay()
 		return true;
 	}
 
+	@ConfigItem(
+			keyName = "superiornotification",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerConfig.java]: Fix your indentation to follow other annotations in class

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 2]
```diff
@@ -229,6 +234,12 @@ public void onChatMessage(ChatMessage event)
 			return;
 		}
 
+		if (chatMsg.equals(CHAT_SUPERIOR_MESSAGE) && config.showSuperiorNotification())
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: put the boolean check before the equality check

File: runelite-client/src/test/java/net/runelite/client/plugins/slayer/SlayerPluginTest.java [hunk 3]
```diff
@@ -70,6 +74,18 @@
 	@Bind
 	ItemManager itemManager;
 
+	@Mock
```
Comment (deathbeam) [inline runelite-client/src/test/java/net/runelite/client/plugins/slayer/SlayerPluginTest.java]: Wouldn't this work if you just mock bind Notifier?

**Comment by xaviergmail** [comment]
Here you go

**Comment by Adam-** [comment]
It would be better to have a test added for this.

**Comment by xaviergmail** [comment]
How do you see this being tested?

**Pull request merged**